src/test/java/io/xstefank/wildfly/bot/utils/JsonHandler.java

```diff
@@ -0,0 +1,83 @@
+package io.xstefank.wildfly.bot.utils;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+import java.io.File;
+import java.io.IOException;
+
+/**
+ * Class responsible for handling and changing properties of the JSON template file.
+ */
+public class JsonHandler {
+
+    private static JsonNode file;
+
+    /**
+     * Constructor for the JsonHandler class.
+     *
+     * @param fileName name of the JSON file located in 'src/test/resources/' folder. The name must contain the .json suffix.
+     * @throws IOException if an error occurred during the JSON file obtaining.
+     */
+    public JsonHandler(String fileName) throws IOException {
+        ObjectMapper objectMapper = new ObjectMapper();
+        file = objectMapper.readTree(new File("src/test/resources/" + fileName));
+    }
+
+    /**
+     * Sets the action (state) of the GitHub PR.
+     *
+     * @param action the action to set. See {@link Actions}.
+     */
+    public void setAction(Actions action) {
+        ((ObjectNode) file).put("action", action.getValue());
+    }
+
+    /**
+     * Sets the Title of the GitHub PR.
+     *
+     * @param title the title to set.
+     */
+    public void setPullRequestTitle(String title) {
+        ((ObjectNode) file.get("pull_request")).put("title", title);
+    }
+
+    /**
+     * Sets the Body (description) of the GitHub PR.
+     *
+     * @param body the body to set.
+     */
+    public void setPullRequestBody(String body) {
+        ((ObjectNode) file.get("pull_request")).put("body", body);
+    }
+
+    /**
+     * Set the amount of commits in the GitHub PR.
+     *
+     * @param quantity the amount of commits to set.
+     */
+    public void setCommitQuantity(int quantity) {
+        ((ObjectNode) file.get("pull_request")).put("commits", quantity);
+    }
+
+    /**
+     * Returns the SHA of the first commit in the GitHub PR.
+     *
+     * @return the SHA of the first commit in the GitHub PR.
+     */
+    public String getCommitSHA() {
+        return file.get("pull_request").get("head").get("sha").textValue();
+    }
+
+    /**
+     * Returns the JSON file using the {@link JsonNode#toPrettyString()}.
+     * The {@code toString} method returns the file as a single line String. The {@code toPrettyString} returns the
+     * file in easily readable way.
+     *
+     * @return the Spring representation of the JSON file.
+     */
+    public String getJsonPrettyString() {
```

Comment:
if there is no method for "nonpretty" string that this might be confusing for the users of this API. Doest the user care if the string returned is "pretty" or not?